### DISCUSSION TITLE
u1 and u2 tx/rr

### DISCUSSION THREAD

**Discussion by bkbartk** [arendst/Tasmota#16737] [Q&A]
Hello,

I am planning to flash this device with Tasmota
https://templates.blakadder.com/lsc_smart_connect_3000272.html
now I completely took it apart, ant found ground and 3.3v.
but there are also 
u1_tx
u1_rx
u2_tx
u2_rx

I would assume u1 is the one to go with, but since my soldering skills aren't that good, I won't have to winder is it is my soldering or the wrong pins.

Does someone know which one would be correct?

**Comment by sfromis** [comment]
Using the u1 pins is a pretty good guess, but I'd not really expect someone with "hard knowledge" to pop up. You could try booting with a temporary hand-held connection on the tx pin. Right after power up, the ESP8266 does send a few lines of boot messages.

**Reply by barbudor** [reply]
Have you found gpio0?
You need that to enable the flashing mode

**Comment by bkbartk** [comment]
Thnx. I tried some stuff. But completely pulled the v1_tx an v2_tx wires from the board. So that was not a success. 
also I couldn’t solder in ground. Tin wouldn’t stick. 
Do I’m going to get me another one and try to use alligator clips for tx an rx. 
gpio00 is one of the pins in esp8266. I have to read the docs to find which one. For as far as I can see. It’s not wired on the board itself.

**Reply by sfromis** [reply]
Soldering to the gpio0 pin on the ESP8266 takes really good soldering skills. You could also touch a pin to it, does not matter if you hit either of the neighbor pins at the same time. Many devices will have a IO0 pad somewhere.....

Maybe use the broken board to practice your soldering skills ....

**Comment by bkbartk** [comment]
> You could also touch a pin to it, does not matter if you hit either of the neighbor pins at the same time
I know,
but in the end, I couldn't get it to work, couldn't get any data out of any tx, So maybe I'll try again later, but for now I ordered some bulbs from another brand hoping they will be easier to flash.

**Comment by bkbartk** [comment]
to give an answer to the question.
the chip actually isn't esp but bekely something so Tasmota wont run anyway.
I bought another lamp and this one has the same chip
but on a WB2L board
https://developer.tuya.com/en/docs/iot/wb2l-datasheet?id=K9duegc9bualu
afaiks I can replace this with
https://templates.blakadder.com/DMP-L1.html
and for the bekely cpu without WB2L, I found software for it, but I'm not sure I'm willing to use that yet.

I actually think a lot o tuya bulbs on the blackadder are now using a non esp chip. because for me this year this was the case 2 out of 2.

**Reply by sfromis** [reply]
It is correct that Tuya devices by now do not have Espressif chips, unless still older designs.
